src/doLifeCycle.js

```diff
@@ -0,0 +1,84 @@
+class LifeCycle {
+
+  transform = value => value;
+  beforeUpdate = [];
+  afterUpdate = [];
+
+  constructor(main, nextValue, opts = {}) {
+    this.main = main;
+    this.nextValue = nextValue;
+
+    this.ensureArgs(opts);
+  }
+
+  ensureArgs = opts => {
+    if (typeof opts === 'function') {
+      this.transform = opts;
+    } else if (typeof opts === 'object') {
+      if (opts.beforeUpdate || opts.afterUpdate) {
+        if (opts.transform) this.transform = opts.transform;
+        this.beforeUpdate = this.ensureArray(opts.beforeUpdate);
+        this.afterUpdate = this.ensureArray(opts.afterUpdate);
+      }
+    } else throw new Error('Illegal options supplied to cycle.');
+  }
+
+  ensureArray = item => [].concat(item);
+
+  ensureUpdatesAreFuncs = () => {
+    const { beforeUpdate, afterUpdate } = this;
+    if (
+      !beforeUpdate.every(func => typeof func === 'function') ||
```

Comment:
This seems like a really good use for typescript-- wouldn't be hard to switch over